### PR TITLE
Deduplicate docs between global and non-global full backward hooks

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119708
* #114970

### CODE REVIEW

File: torch/nn/modules/module.py [hunk 1]
```diff
@@ -281,22 +281,12 @@ def register_module_full_backward_pre_hook(
         This adds global state to the `nn.module` module
         and it is only intended for debugging/profiling purposes.
 
-    The hook will be called every time the gradients for the module are computed.
-    The hook should have the following signature::
+    Hooks registered using this function function in the same way as those
```
Comment (albanD) [inline torch/nn/modules/module.py:284]: function function is grammatically correct but hard to read. 'behaves" might be better

File: torch/nn/modules/module.py [hunk 2]
```diff
@@ -281,22 +281,12 @@ def register_module_full_backward_pre_hook(
...
+    registered by :meth:`torch.nn.Module.register_backward_pre_hook`.
```
Comment (albanD) [inline torch/nn/modules/module.py:285]: The full hooks should be the ones that we want everyone to use. We shouldn't document them based on these other ops we don't want users to use.

If you want to dedup, you can update the non-full hooks to say not to use them, point to the full version for full doc and mention the difference.

File: torch/nn/modules/module.py (hunk 1 above)
Comment (soulitzer) [reply torch/nn/modules/module.py:284]: Good point haha

File: torch/nn/modules/module.py (hunk 2 above)
Comment (soulitzer) [reply torch/nn/modules/module.py:285]: Whoops typo

Comment (soulitzer) [reply torch/nn/modules/module.py]: > If you want to dedup, you can update the non-full hooks to say not to use them, point to the full version for full doc and mention the difference.

Yeah for non-full global hooks I think we already do that today.

File: torch/nn/modules/module.py [hunk 3]
```diff
@@ -318,26 +308,12 @@ def register_module_full_backward_hook(
...
-    subsequent computations. :attr:`grad_input` will only correspond to the inputs given
-    as positional arguments and all kwarg arguments will not appear in the hook. Entries
-    in :attr:`grad_input` and :attr:`grad_output` will be ``None`` for all non-Tensor
-    arguments.
-
-    For technical reasons, when this hook is applied to a Module, its forward function will
-    receive a view of each Tensor passed to the Module. Similarly the caller will receive a view
-    of each Tensor returned by the Module's forward function.
+    Hooks registered using this function behave in the same way as those
+    registered by :meth:`torch.nn.Module.register_full_backward_hook`.
```
[the remark this answers is not in the document]
Comment (soulitzer) [reply torch/nn/modules/module.py:312]: "Register a backward pre-hook common to all the modules." is already mentioned at the beginning of the doc

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**